### ISSUE TITLE
OpenGL rendering issues with Linux and some windows graphic cards

### ISSUE THREAD

**Issue by nicolas-f** [Ifsttar/I-Simpa#185]
Surface receivers, legend and particles may not render correctly. This source of the problem could be an incorrect OpenGL calls.

**Comment by lonp** [comment]
Problems with release 1.3.4 and Intel HD Graphics. So I came back to previous version (1.2.3) that gave me this message "Error: Calculation .exe file not found."

**Comment by wbinek** [comment]
I don't know if it is the same issue, but I also have a display problems. This applies to both my builds and the official ones. I've tried building with wxWidgets 3.1.1 but it didn't solve the problem. I believe the problems started after one of the Windows 10 updates.

On some systems the background and currently selected faces or faces with default material or are black. Additionality sometimes the 3DViewer doesn't refresh automatically - to force refresh I need to click on any element from project tree.

**Comment by nicolas-f** [comment]
Yes. We have some 3D issues with Windows 10 (have to select window 7 compatibility mode).

This is some wrong OpenGL calls. We are currently working at fixing this issue by integrating a 3D engine into I-Simpa (Ogre or Irrlicht engine).

**Comment by wbinek** [comment]
Hi!
I managed to solve my problems with 3D view not refreshing by adding `
this->Refresh();` after `GetEventHandler()->ProcessEvent(ptevt);` in `OpenGlViewer::OnTimer()` method. I hope it will help at least as a temporary solution to some of the problems :)

**Comment by nicolas-f** [comment]
Thanks, we will patch I-Simpa soon with your contribution :)

**Comment by Chum4k3r** [comment]
@wbinek and @nicolas-f would you please specify how to add these lines to the main code? I'm facing the "no refresh" issue on a Win10 x64 with Intel HD + ATI Radeon graph cards. Apparently everything else seems fine, but this auto refreshing, wich is very annoying  when dealing with projects.

**Comment by wbinek** [comment]
Take a look at:
https://github.com/wbinek/I-Simpa/commit/09171a804b91bb849caf768c6becf146eb088ff8

**Comment by nicolas-f** [comment]
VulkanSceneGraph integration with wxwidgets

https://github.com/deadcast2/vsgwx
